### PR TITLE
Address `kid`

### PR DESCRIPTION
**Pull request by OR13**
Attempt to close remaining `kid` issues: #30 and #117

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/pull/163.html" title="Last updated on Oct 6, 2023, 10:06 PM UTC (9b37beb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-jose-cose/163/4677a03...9b37beb.html" title="Last updated on Oct 6, 2023, 10:06 PM UTC (9b37beb)">Diff</a>

### CODE REVIEW

**Comment by vdods** [comment]
Looks great!

Though I think there might be a typo in the links to "DID URL" -- they point to "relative DID URL":
```
When the <a data-cite="VC-DATA-MODEL#dfn-issuers">issuer</a> is identified as a <a data-cite="DID-CORE#did-subject">DID Subject</a>, 
        the <a href="#kid">kid</a> MUST be an absolute <a data-cite="DID-CORE#relative-did-urls">DID URL</a>.
```
I believe there are a few places in the changes that have this typo.

If this is intentional, then please ignore this comment.

**Comment by OR13** [comment]
It is intentional, I found that section most helpful in explaining the difference.

**Review by decentralgabe — APPROVED** [review]

File: index.html [hunk 1]
```diff
@@ -384,11 +384,94 @@ <h2>Key Discovery</h2>
...
+{
+  "alg": "ES384",
+  "kid": "did:example:abc#key-456
+}
+</pre>
+
+      <!-- REGULAR URLS via "issuer" and "holder" -->
+      <p>
+        When the <a data-cite="VC-DATA-MODEL#dfn-issuers">issuer</a> is identified as a [[URL]], 
+        the <a href="#kid">kid</a> MUST be an absolute [[URL]] to a verification method listed in a controller document.
```
Comment (selfissued) [inline index.html:425]: Can "controller document" be a reference to a defined term?

**Review by selfissued — APPROVED** [review]
LGTM, modulo my non-blocking comment about what "controller document" should refer to.

File: index.html (hunk 1 above)
Comment (OR13) [reply index.html:425]: sadly not yet, this is due to the dialog regarding data integrity defining controller document.

Comment (OR13) [reply index.html:425]: See: https://github.com/w3c/vc-jose-cose/issues/160

**Review by selfissued — APPROVED** [review]

File: index.html [hunk 2]
```diff
@@ -384,11 +384,97 @@ <h2>Key Discovery</h2>
...
+        is identified as a <a data-cite="DID-CORE#did-subject">DID Subject</a>, 
+        the <a href="#kid">kid</a> MUST be an absolute <a data-cite="DID-CORE#relative-did-urls">DID URL</a>.
+      </p>
+<pre class="example" title="A holder identified by a DID">
+{
+  "holder": "did:example:abc"
+  // ...
+}
+</pre>
+<pre class="example" title="An kid as an absolute DID URL">
```
Comment (TallTed) [inline index.html:417]:
```suggestion
<pre class="example" title="A kid as an absolute DID URL">
```

File: index.html [hunk 3]
```diff
@@ -384,11 +384,97 @@ <h2>Key Discovery</h2>
...
+{
+  "alg": "ES384",
+  "kid": "https://university.example/issuers/565049#key-123
+}
+</pre>
+
+      <!-- REGULAR URLS via "iss" -->
+      
+      <p>
+        When <a href="iss">iss</a> is a present, and is a [[URL]],
```
Comment (TallTed) [inline index.html:468]:
```suggestion
        When <a href="iss">iss</a> is present, and is a [[URL]], 
```

File: index.html [hunk 4]
```diff
@@ -384,11 +384,97 @@ <h2>Key Discovery</h2>
...
+
+      <!-- REGULAR URLS via "iss" -->
+      
+      <p>
+        When <a href="iss">iss</a> is a present, and is a [[URL]], 
+        the <a href="#kid">kid</a> MUST match a key discovered via <a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-00#name-jwt-issuer-metadata">JWT Issuer Metadata Request</a>
+      </p>
+
+      <p class="issue" title="(AT RISK) Feature depends on demonstration of independent implementations">
+        This normative statement depends on a -00 IETF OAUTH WG Adopted draft.
```
Comment (TallTed) [inline index.html:473]:
```suggestion
        This normative statement depends on a
<a href="https://datatracker.ietf.org/doc/html/draft-ietf-oauth-sd-jwt-vc-00#name-jwt-issuer-metadata">-00 IETF OAUTH WG Adopted draft</a>
```

**Comment by TallTed** [comment]
@OR13 -- Please let me know if you need a PR from me for the above minor tweaks to this. I would have preferred to submit them as change requests before this PR was merged, but time travel remains tragically unavailable.

**Comment by OR13** [comment]
@TallTed  please raise a PR to make these suggestions.

**Comment by TallTed** [comment]
Done. See #171.

**Pull request merged**